### PR TITLE
Enable switching between XOR metric and Hamming distance for testing

### PR DESCRIPTION
**Pull request by neonknight64**
## Description
- These changes allow the the distance metric used to determine the distance between nodes to be switched between the XOR metric and the Hamming distance. The default is still the XOR metric.
- The Hamming distance seems to produce more uniform connections between different nodes, reducing the clustering of nodes that are not able to communicate with each other.

## Motivation and Context
The XOR metric seems to cause clustering of nodes that are not able to communicate. Hamming distance seems to improve the clustering issue. This PR allows the switching between the two distance algorithms so that the Hamming distance can be tested.

## How Has This Been Tested?
Existing tests were updated to remain functional with changes to the distance algorithm. Some tests that could not be changed have the results of both distance algorithms.

## Types of changes

* [x] Bug fix (non-breaking change which fixes an issue)
* [x] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Feature refactor (No new feature or functional changes, but performance or technical debt improvements)
* [ ] New Tests
* [ ] Documentation

## Checklist:

* [x] I'm merging against the `development` branch.
* [x] I ran `cargo-fmt --all` before pushing.
* [x] I have squashed my commits into a single commit.
* [ ] My change requires a change to the documentation.
* [ ] I have updated the documentation accordingly.
* [ ] I have added tests to cover my changes.

### CODE REVIEW

**Review by sdbondi — COMMENTED** [review]
Because this change is still being debated/tested and will cause a large network reshuffle, I'd like to be able to switch between them easily.

```rust
pub type NodeDistance = XorDistance; // or HammingDistance
```
HammingDistance can use XorDistance internally.

Then we can merge the PR and switch over with a one liner when/if we are happy to make the change.

File: comms/src/peer_manager/node_id.rs [hunk 1]
```diff
@@ -42,75 +42,133 @@ use tari_crypto::tari_utilities::{
 const NODE_ID_ARRAY_SIZE: usize = 13; // 104-bit as per RFC-0151
 type NodeIdArray = [u8; NODE_ID_ARRAY_SIZE];
 
+pub type NodeDistance = HammingDistance; // or XorDistance
```
Comment (sdbondi) [inline comms/src/peer_manager/node_id.rs]: Think we should merge this PR and then make another PR to switch it if/when we have experimented a bit with the HammingDistance metric

```suggestion
pub type NodeDistance = XorDistance; // or HammingDistance
```

**Review by sdbondi — COMMENTED** [review]
Thanks for the changes - think let's keep it as XorDistance so that we can merge this PR and allow us to experiment if/when we switch

File: comms/src/peer_manager/node_id.rs (hunk 1 above)
Comment (neonknight64) [reply comms/src/peer_manager/node_id.rs]: Change applied.

**Review by sdbondi — APPROVED** [review]
Thanks - switching between the two works great! Now we can experiment a bit 🔬
